x/tx/signing/direct_aux/direct_aux.go

```diff
@@ -0,0 +1,51 @@
+package direct_aux
+
+import (
+	"context"
+	"fmt"
+
+	"google.golang.org/protobuf/proto"
+
+	signingv1beta1 "cosmossdk.io/api/cosmos/tx/signing/v1beta1"
+	txv1beta1 "cosmossdk.io/api/cosmos/tx/v1beta1"
+	"cosmossdk.io/x/tx/signing"
+)
+
+// SignModeHandler is the SIGN_MODE_DIRECT_AUX implementation of signing.SignModeHandler.
+type SignModeHandler struct{}
+
+var _ signing.SignModeHandler = SignModeHandler{}
+
+// Mode implements signing.SignModeHandler.Mode.
+func (h SignModeHandler) Mode() signingv1beta1.SignMode {
+	return signingv1beta1.SignMode_SIGN_MODE_DIRECT_AUX
+}
+
+// GetSignBytes implements signing.SignModeHandler.GetSignBytes.
+func (SignModeHandler) GetSignBytes(
+	_ context.Context, signerData signing.SignerData, txData signing.TxData) ([]byte, error) {
+
+	// replicate https://github.com/cosmos/cosmos-sdk/blob/4a6a1e3cb8de459891cb0495052589673d14ef51/x/auth/tx/builder.go#L142
+	feePayer := txData.AuthInfo.Fee.Payer
+	if feePayer == "" {
+		signers, err := signing.NewGetSignersContext(signing.GetSignersOptions{}).GetSigners(txData.Body)
```

Comment:
How hard would it be to pass `GetSignersContext` into this handler, instead of creating a new instance here?

Reply:
@amaurym I'm using the code you suggested, bu getting this back from `GetSigners`

```
        	            	actual  : "no cosmos.msg.v1.signer option found for message google.protobuf.Any"
```

Have I done something wrong with creating the SignersContext or something? It's not really clear to me from the error.  I'll keep continue reverse engineering it but if you have any insight it'd be helpful.

Reply:
Oh yeah, sorry, my code was incorrect. You need to unpack the any before calling GetSigners.

```go
var signers []string
for _, msgAny := range Body.Msgs {
  msg, err:= anyutil.Unpack(msgAny)
  msgSigners := GetSignersContext.GetSigners(msg)
  signers := append(signers, msgSigners...)
}
```